aimmdb/schemas.py

```diff
@@ -114,6 +120,16 @@ class XASMetadata(pydantic.BaseModel, extra=pydantic.Extra.allow):
     sample_id: Optional[str]
 
 
+class FacilityMetadata(pydantic.BaseModel, extra=pydantic.Extra.allow):
+    name: str
+
+    @pydantic.validator("name")
```

Comment:
I believe this can be implemented using an enum rather than a validation method. Where possible, statically-analyzable validation is preferred.

Reply:
Thanks for the input Dan, and sorry for not responding earlier. I'm reworking the FEFF schemas and will definitely incorporate your suggestion.

For the statistical analysis, you're saying we could write a separate function that allows us to enumerate the metadata? Would one application be determining the number of uploaded samples?

With that in mind, should I replace all the validation methods with enumerations?

Reply:
I’m not sure without looking at the validation methods in question. If the thing they are validating is, "Is this value in this set?" then yes.